integration-libs/epd-visualization/components/visual-viewer/visual-viewer.service.ts

```diff
@@ -0,0 +1,1496 @@
+import {
+  ChangeDetectorRef,
+  ElementRef,
+  EventEmitter,
+  Input,
+  Output,
+  Injectable,
+} from '@angular/core';
+import { BehaviorSubject, Observable, of } from 'rxjs';
+import {
+  EpdVisualizationConfig,
+  EpdVisualizationInnerConfig,
+  Ui5Config,
+  VisualizationApiConfig,
+} from '@spartacus/epd-visualization/root';
+import { SceneNodeToProductLookupService } from '@spartacus/epd-visualization/root';
+import { VisualizationLookupService } from '@spartacus/epd-visualization/root';
+
+// @ts-ignore
+import * as ui5 from '@sapui5/ts-types-esm';
+
+import type Core from 'sap/ui/core/Core';
+import type { CSSColor } from 'sap/ui/core/library';
+
+import type Scene from 'sap/ui/vk/Scene';
+import type Viewport from 'sap/ui/vk/Viewport';
+import type ContentConnector from 'sap/ui/vk/ContentConnector';
+import type AnimationPlayer from 'sap/ui/vk/AnimationPlayer';
+import type DrawerToolbar from 'sap/ui/vk/DrawerToolbar';
+import type ViewStateManager from 'sap/ui/vk/ViewStateManager';
+type ViewManager = any;
+type NodeRef = any;
+import type UIArea from 'sap/ui/core/UIArea';
+import type ContentResource from 'sap/ui/vk/ContentResource';
+
+import { NavigationMode } from './models/navigation-mode';
+import { SelectionMode } from './models/selection-mode';
+import { NodeContentType } from './models/node-content-type';
+import { SceneLoadState } from './models/scene-load-state';
+import { LoadedSceneInfo, SceneLoadInfo } from './models/scene-load-info';
+import { ContentType } from '@spartacus/epd-visualization/root';
+import NodeHierarchy from 'sap/ui/vk/NodeHierarchy';
+import { catchError, filter, mergeMap, shareReplay } from 'rxjs/operators';
+import { VisualizationInfo } from '@spartacus/epd-visualization/root';
+import {
+  VisualizationLoadInfo,
+  VisualizationLoadStatus,
+  VisualizationLookupResult,
+} from './models/visualization-load-info';
+import { ZoomTo } from './models/zoom-to';
+
+interface VisualContentChangesFinishedEvent {
+  content: any;
+  failureReason: any;
+}
+
+interface VisualContentLoadFinishedEvent {}
+
+@Injectable({
+  providedIn: 'any',
+})
+export class VisualViewerService {
```

Comment:
This service's implementation seems way too big to reason about. Why is it so?
Do you think we could distinguish a few separate responsibilities of this service and split them out into a few smaller services?

Reply:
I can look at breaking it down a bit.

Reply:
Do you still have plans to break it down? 
If not in this iteration (due to time constrains), when do you think you could prioritize it?